### PR TITLE
Misleading docs on use of regexp

### PR DESCRIPTION
**Pull request by lukehinds**
README was using `subject`, with RegExp which requires
an exact match

Signed-off-by: Luke Hinds <lhinds@redhat.com>

### CODE REVIEW

**Review by hectorj2f — COMMENTED** [review]
@lukehinds I think we have one more instance in the README.md: 
https://github.com/sigstore/policy-controller/pull/210/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R245

I also think we might want to change the occurrences such as `subject: ".*subject.*"` which are in some testing examples.

File: test/testdata/policy-controller/valid/valid-policy.yaml [hunk 1]
```diff
@@ -75,7 +75,7 @@ spec:
   - keyless:
       identities:
       - issuer: "issue-details1"
-        subject: ".*subject.*"
+        SubjectRegExp: ".*subject.*"
```
Comment (hectorj2f) [inline test/testdata/policy-controller/valid/valid-policy.yaml:78]: Yes, this should be subjectRegExp. That is the reason why tests are failing.

File: test/testdata/policy-controller/valid/v1beta1-valid-policy.yaml [hunk 2]
```diff
@@ -74,7 +74,7 @@ spec:
   - keyless:
       identities:
       - issuer: "issue-details1"
-        subject: ".*subject.*"
+        SubjectRegExp: ".*subject.*"
```
[the remark this answers is not in the document]
Comment (lukehinds) [reply test/testdata/policy-controller/valid/v1beta1-valid-policy.yaml:77]: Oh sorry , case name!

**Review by hectorj2f — APPROVED** [review]
lgtm, Thanks @lukehinds :)

**Review by vaikas — APPROVED** [review]
Thanks @lukehinds

**Pull request merged**